pkg/server/infrastructure/datastore/mysql/mysql.go

```diff
@@ -47,40 +64,249 @@ func New(ctx context.Context, cfg datastore.Config) (datastore.DataStore, error)
-	return true, fmt.Errorf("method to be implemented")
+	if entity.PrimaryKey() == "" {
+		return false, datastore.ErrPrimaryEmpty
+	}
+	if entity.TableName() == "" {
+		return false, datastore.ErrTableNameEmpty
+	}
+
+	if dbExist := m.client.First(entity); dbExist.Error != nil {
+		if errors.Is(dbExist.Error, gorm.ErrRecordNotFound) {
+			return false, nil
+		}
+		return false, datastore.NewDBError(dbExist.Error)
+	}
+
+	return true, nil
 }
 
 // Delete delete data
 func (m *mysql) Delete(ctx context.Context, entity datastore.Entity) error {
-	return fmt.Errorf("method to be implemented")
+	if entity.PrimaryKey() == "" {
+		return datastore.ErrPrimaryEmpty
+	}
+	if entity.TableName() == "" {
+		return datastore.ErrTableNameEmpty
+	}
+	// check entity is exist
+	if err := m.Get(ctx, entity); err != nil {
+		return err
+	}
+
+	if dbDelete := m.client.Unscoped().Model(entity).Delete(entity); dbDelete.Error != nil {
+		klog.Errorf("delete document failure %w", dbDelete.Error)
+		return datastore.NewDBError(dbDelete.Error)
+	}
+
+	return nil
+}
+
+func _applyFilterOptions(clauses []clause.Expression, filterOptions datastore.FilterOptions) []clause.Expression {
+	for _, queryOp := range filterOptions.Queries {
+		clauses = append(clauses, clause.Like{
+			Column: strings.ToLower(queryOp.Key),
+			Value:  fmt.Sprintf("%%%s%%", queryOp.Query),
+		})
+	}
+	for _, queryOp := range filterOptions.In {
+		values := make([]interface{}, len(queryOp.Values))
+		for i, v := range queryOp.Values {
+			values[i] = v
+		}
+		clauses = append(clauses, clause.IN{
+			Column: strings.ToLower(queryOp.Key),
+			Values: values,
+		})
+	}
+	for _, queryOp := range filterOptions.IsNotExist {
+		clauses = append(clauses, clause.Eq{
+			Column: strings.ToLower(queryOp.Key),
+			Value:  "",
+		})
+	}
+	return clauses
 }
 
 // List list entity function
 func (m *mysql) List(ctx context.Context, entity datastore.Entity, op *datastore.ListOptions) ([]datastore.Entity, error) {
-	return nil, fmt.Errorf("method to be implemented")
+	if entity.TableName() == "" {
+		return nil, datastore.ErrTableNameEmpty
+	}
+	var (
+		clauses []clause.Expression
+		exprs   []clause.Expression
+		limit   int
+		offset  int
+	)
+	if op != nil && op.PageSize > 0 && op.Page > 0 {
+		limit = op.PageSize
+		offset = op.PageSize * (op.Page - 1)
+		clauses = append(clauses, clause.Limit{
+			Limit:  &limit,
+			Offset: offset,
+		})
+	}
+	for k, v := range entity.Index() {
+		exprs = append(exprs, clause.Eq{
+			Column: strings.ToLower(k),
+			Value:  v,
+		})
+	}
+	if op != nil {
+		exprs = _applyFilterOptions(exprs, op.FilterOptions)
+	}
+	if len(exprs) > 0 {
+		clauses = append(clauses, clause.Where{
+			Exprs: exprs,
+		})
+	}
+	if op != nil && op.SortBy != nil {
+		var sortOption []clause.OrderByColumn
+		for _, v := range op.SortBy {
+			sortOption = append(sortOption, clause.OrderByColumn{
+				Column: clause.Column{
+					Name: strings.ToLower(v.Key),
+				},
+				Desc: (v.Order == -1),
```

Comment:
Unwrap the parentheses. And use `SortOrderDescending`. This may cause loop importing. Extract SortOrderDescending to `datastore/option`.

Reply:
Sorry, I forgot to use `datastore.SortOrderDescending`. I use it in place of `-1`, I thing it will not create loop importing error.